HttplugFactory.php

```diff
@@ -95,11 +93,14 @@ protected function evaluateCondition($condition)
             $evaluatedCondition = true;
 
             // Immediately stop execution if the condition is false
-            for ($i = 0; $i < count($condition) && false !== $evaluatedCondition; ++$i) {
-                $evaluatedCondition &= $this->evaluateCondition($condition[$i]);
```

Comment:
I would not remove this as it is a wonderful solution for sparing that extra if you added.